pkg/rabbitmq/handlers.go

```diff
@@ -87,8 +87,9 @@ func (h *PasswordChangeHandler) Handle(ctx context.Context, d amqp.Delivery) err
 		log.Debugf("password change: skipping key parameters (incomplete pair) for TwakeID: %s", msg.TwakeID)
 	}
 
-	log.Debugf("password change: retrieving instance for domain: %s", msg.Domain)
-	inst, err := lifecycle.GetInstance(msg.Domain)
+	userDomain := msg.TwakeID + "." + msg.Domain
+	log.Debugf("password change: retrieving instance for domain: %s", userDomain)
+	inst, err := lifecycle.GetInstance(userDomain)
```

Comment:
You need the fully qualified domain name here (i.e. FQDN) which might not be `<twake_id>.<domain>` as the cloudery can change that before creating the instance.